src/frontend/src/app/assets/assets.component.ts

```diff
@@ -7,14 +7,35 @@ import { AssetsService, AlertService } from '../services/index';
   styleUrls: ['./assets.component.css']
 })
 export class AssetsComponent implements OnInit {
+  public limit: Number = 0;
+  public offset: Number = 0;
   assets = [];
   assetsReadingsData = [];
+  asset_code;
```

Comment:
Rename it to something like asssetCodeField because it is going to hold a html. Also I feel it can be moved in   getAssetReading() method as a local variable.